src/status.c

```diff
@@ -0,0 +1,104 @@
+		if (code[1] < 0x80) {
+			format_status(p, "Note length: %d, velocity: %d/7, release: %d/15", code[0], (code[1] >> 4) & 0x7, code[1] & 0xF);
+		} else {
+			format_status(p, "Note length: %d", code[0]);
+		}
+	} else if (code[0] < 0xC8) {
+		BYTE note = (code[0] & 0x7F);
+		format_status(p, "Note %s%c", notes[note % 12], '1' + note / 12);
+	} else if (code[0] == 0xC8) {
+		format_status(p, "Hold");
+	} else if (code[0] == 0xC9) {
+		format_status(p, "Rest");
+	} else if (code[0] < 0xE0) {
+		format_status(p, "Percussion %d", code[0] - 0xC9);
+	} else {
+		switch (code[0]) {
+			case 0xE0: format_status(p, "Instrument %d", code[1]); break;
+			case 0xE1: format_status(p, "Pan %d", code[1]); break;
+			case 0xE2: format_status(p, "Pan (duration: %d, pan: %d)", code[1], code[2]); break;
+			case 0xE3:
+				format_status(p, "Enable vibrato (delay: %d, freq: %d, range: \xB1%d%s)",
+					code[1],
+					code[2],
+					code[3] <= 0xF0 ? code[3] : code[3] - 0xF0,
+					code[3] <= 0xF0 ? "/256 semitones" : " semitones");
+				break;
+			case 0xE4: format_status(p, "Disable vibrato"); break;
+			case 0xE5: format_status(p, "Master volume %d/255", code[1]); break;
+			case 0xE6: format_status(p, "Master volume (duration: %d, volume: %d/255)", code[1], code[2]); break;
+			case 0xE7: format_status(p, "Tempo %d", code[1]); break;
+			case 0xE8: format_status(p, "Tempo (duration: %d, tempo: %d)", code[1], code[2]); break;
+			case 0xE9: format_status(p, "Global transpose %d", (signed char)code[1]); break;
+			case 0xEA: format_status(p, "Channel transpose %d", (signed char)code[1]); break;
+			case 0xEB: format_status(p, "Enable tremolo (delay: %d, freq: %d, amp: %d)", code[1], code[2], code[3]); break;
+			case 0xEC: format_status(p, "Disable tremolo"); break;
+			case 0xED: format_status(p, "Channel volume %d/255", code[1]); break;
+			case 0xEE: format_status(p, "Channel volume (duration: %d, volume: %d/255)", code[1], code[2]); break;
+			case 0xEF: format_status(p, "Call subroutine %d, %d %s", code[1] | code[2] << 8, code[3], code[3] == 1 ? "time" : "times"); break;
+			case 0xF0: format_status(p, "Set vibrato attack %d", code[1]); break;
+			case 0xF1: format_status(p, "Enable portamento (delay: %d, duration: %d, range: +%d)", code[1], code[2], code[3]); break;
+			case 0xF2: format_status(p, "Enable portamento (delay: %d, duration: %d, range: -%d)", code[1], code[2], code[3]); break;
+			case 0xF3: format_status(p, "Disable portamento"); break;
+			case 0xF4: format_status(p, "Finetune %d/256 semitones", code[1]); break;
+			case 0xF5:
+				format_status(p, "Enable echo (channels: %c%c%c%c%c%c%c%c, left volume: %d/255, right volume: %d/255)",
+					'0' + (code[1] & 1),
+					'0' + ((code[1] >> 1) & 1),
+					'0' + ((code[1] >> 2) & 1),
+					'0' + ((code[1] >> 3) & 1),
+					'0' + ((code[1] >> 4) & 1),
+					'0' + ((code[1] >> 5) & 1),
+					'0' + ((code[1] >> 6) & 1),
+					'0' + ((code[1] >> 7) & 1),
+					code[2],
+					code[3]);
```

Comment:
I believe left volume and right volume are more `signed char`s. These numbers are written directly to DSP registers and they're out of 128. (Docs: [fullsnes](https://problemkaputt.de/fullsnes.htm#snesapudspechoregisters), [S-DSP REGISTERS section of anomie's DSP documentation](https://www.romhacking.net/documents/191/)

Reply:
Changed this to signed values without specifying phase inversions. Hopefully that should be intuitive